enterprise/backend/test/metabase_enterprise/semantic_search/embedding_test.clj

```diff
@@ -86,3 +92,140 @@
           batches (#'embedding/create-batches 5 #'embedding/count-tokens texts)]
       ;; Should skip the long text and batch the short ones
       (is (= [["Short" "Also short"]] batches)))))
+
+(defn- float-vectors-approx=
+  "Compare two vectors of floats for approximate equality within a given tolerance."
+  ([v1 v2]
+   (float-vectors-approx= v1 v2 0.0001))
+  ([v1 v2 tolerance]
+   (and (= (count v1) (count v2))
+        (every? #(< (Math/abs ^double %) tolerance)
+                (map - v1 v2)))))
+
+(defn- encode-floats-to-base64
+  "Helper function to base64 encode a vector of floats using little-endian byte order.
+   Inverse of [[metabase-enterprise.semantic-search.embedding/extract-base64-response-embeddings]] "
+  [float-vector]
+  (let [byte-count (* (count float-vector) 4)
+        buffer (doto (ByteBuffer/allocate byte-count)
+                 (.order ByteOrder/LITTLE_ENDIAN))]
+    (doseq [f float-vector]
+      (.putFloat buffer (float f)))
+    (.encodeToString (Base64/getEncoder) (.array buffer))))
+
+(deftest test-extract-base64-response-embeddings
```

Comment:
Unrelated, but I added some tests for this function since I needed to encode mock responses to test `get-embedding`